cranelift-codegen/meta/src/shared/legalize.rs

```diff
@@ -612,6 +617,18 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
         }
     }
 
+    for &ty in &[I8, I16] {
+        widen.legalize(
+            def!(brz.ty(x, ebb, vararg)),
+            vec![def!(a = uextend.I32(x)), def!(brz(a, ebb, vararg))],
+        );
+
+        widen.legalize(
+            def!(brnz.ty(x, ebb, vararg)),
+            vec![def!(a = uextend.I32(x)), def!(brnz(a, ebb, vararg))],
+        );
+    }
```

Comment:
@julian-seward1, if this was what you were concerned about when you mentioned "partial register stalls," I think that is avoided here because of `uextend` which maps to a `MOV` with zero-extension. If you were concerned about using 16-bit registers on x86 in general, then it seems that is a bigger issue (should Cranelift then have an `I16` type?).